rio_tiler/mosaic/methods/defaults.py

```diff
@@ -114,17 +114,86 @@ class StdevMethod(MosaicMethodBase):
     def __init__(self, enforce_data_type=True):
         """Overwrite base and init Stdev method."""
         super(StdevMethod, self).__init__()
+        self.enforce_data_type = enforce_data_type
         self.tile = []
 
     @property
     def data(self):
         """Return data and mask."""
         if self.tile:
             tile = numpy.ma.std(numpy.ma.stack(self.tile, axis=0), axis=0)
+            tile = tile.astype(self.tile[0].dtype) if self.enforce_data_type else tile
             return tile.data, ~tile.mask[0] * 255
         else:
             return None, None
 
     def feed(self, tile):
         """Add data to tile."""
         self.tile.append(tile)
+
+class GeoMedianMethod(MosaicMethodBase):
+
+    def __init__(self, enforce_data_type=True):
+        super(GeoMedianMethod, self).__init__()
+        self.enforce_data_type = enforce_data_type
+        self.tile = []
+
+    @property
+    def data(self):
+        """Return data and mask."""
+        if self.tile:
+            # Convert with rescaling factor to float32 for input to Dale's hdmedians
+            stacked_tile = (numpy.ma.stack(self.tile, axis=1) / 10000).astype(numpy.float32)
+            b, t, y, x = stacked_tile.shape
+            tile = numpy.ma.zeros((b, y, x)).astype(numpy.float32)
+
+            # TODO: How to implement geomedian without introducing too many new dependencies?
+            for xid in range(x):     # for each x-axis
+                for yid in range(y): # for each y-axis
+                    tile[:, yid, xid] = hd.geomedian(stacked_tile[:, :, yid, xid])
```

Comment:
Yeah that's won't be fast 😬 

I wonder if we could vectorize the calculation with numpy ufunc (cc @kylebarron)

Reply:
I agree this won't be fast.

I think the main problem here is that `hd.geomedian` only supports a two-dimensional array as input. So it can't natively broadcast over a higher dimensional array.

Also how does this work with missing data? Does it automatically mask out portions of the array that are undefined?

Reply:
About the missing data, It returns `0s` currently, which you can then convert to `NaNs` _a posteriori_, or directly use the `hd.nangeomedian`, which handles missing data directly (should have used that one for rigor's sake). 

As I said, I believe the n-dimensional (at least 4-dimensional) implementation of their geomedian sits somewhere at Geoscience Australia already developed, as I can see a `geomedian_pcm` (and even a `nangeomedian_pcm`) called [here,](https://github.com/GeoscienceAustralia/burn-mapping/pull/7/files#diff-4b64c64388aca506a037e44b5433d786R17)

That's why I'm reluctant to make a PR on `hdmedians` as I'm pretty certain the work's already been done.

Paper for reference: [https://ieeexplore.ieee.org/stamp/stamp.jsp?arnumber=8004469](https://ieeexplore.ieee.org/stamp/stamp.jsp?arnumber=8004469)

Reply:
That seems to import `hdstats`, which isn't on PyPI... I can't find anything about that package

Reply:
I got confirmation from [Dale Roberts](https://github.com/daleroberts) that `hdstats` will be made available through PyPI in the coming months so we can stall this for now.